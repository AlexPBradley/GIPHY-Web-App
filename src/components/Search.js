import React, { useState, useEffect } from "react";
import getGifs from "../requests/getGifs.js";
import PropTypes from "prop-types";
import "../styles/search.css";

const Search = ({ setGifResults, setResultsCount, setSearchValue }) => {
  const [value, setValue] = useState("");
  const [limitSelected, setLimitSelected] = useState(0);

  useEffect(() => {
    document.getElementById(
      "button-background"
    ).style.transform = `translate(${limitSelected}px)`;
  });

  const handleSubmit = async (event) => {
    event.preventDefault();
    setGifResults(await getGifs(value));
    value
      ? setSearchValue(value.charAt(0).toUpperCase() + value.slice(1))
      : setSearchValue("Trending");
  };

  return (
    <div className="search">
      <form
        className="search-form"
        data-testid="search-form"
        onSubmit={handleSubmit}
      >
        <input
          className="search-input"
          data-testid="search-input"
          type="text"
          placeholder="Search..."
          onChange={(t) => setValue(t.target.value)}
        />
      </form>
      <div className="limitButtons">
        <button
          className="search-button"
          type="button"
          onClick={() => {
            setResultsCount(10);
            setLimitSelected(0);
          }}
        >
          <span id="button-background"></span>
          10
        </button>
        <button
          id="middle"
          className="search-button"
          type="button"
          onFocus={() => {
            setResultsCount(25);
            setLimitSelected(28);
          }}
        >
          25
        </button>
        <button
          className="search-button"
          type="button"
          onClick={() => {
            setResultsCount(50);
            setLimitSelected(55);
          }}
        >
          50
        </button>
      </div>
    </div>
  );
};

export default Search;

Search.propTypes = {
  setGifResults: PropTypes.func,
};
