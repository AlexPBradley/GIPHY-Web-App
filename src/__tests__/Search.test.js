import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Search />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("form renders correctly", () => {
    render(<Search />);

    const searchForm = screen.getByTestId("search-form");

    expect(searchForm).toBeInstanceOf(HTMLFormElement);
  });

  it("input renders correctly", () => {
    render(<Search />);

    const searchForm = screen.getByTestId("search-input");

    expect(searchForm).toBeInstanceOf(HTMLInputElement);
  });

  it("buttons renders correctly", () => {
    render(<Search />);

    const searchButtons = screen.getAllByTestId("search-button");

    expect(searchButtons).toBeInstanceOf(Array);
  });

  it("button background correctly", () => {
    render(<Search />);

    const buttonBackground = screen.getByTestId("button-background");

    expect(buttonBackground).toBeInstanceOf(HTMLSpanElement);
  });
});
