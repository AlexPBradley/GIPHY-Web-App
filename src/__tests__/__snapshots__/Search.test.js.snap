// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form
      class="search-form"
      data-testid="search-form"
    >
      <input
        class="search-input"
        data-testid="search-input"
        placeholder="Search..."
        type="text"
      />
    </form>
    <div
      class="limitButtons"
    >
      <button
        class="search-button"
        type="button"
      >
        <span
          id="button-background"
          style="transform: translate(0px);"
        />
        10
      </button>
      <button
        class="search-button"
        id="middle"
        type="button"
      >
        25
      </button>
      <button
        class="search-button"
        type="button"
      >
        50
      </button>
    </div>
  </div>
</DocumentFragment>
`;
